### DISCUSSION TITLE
Horizontal scroll for back & forward almost works.. (video included)

### DISCUSSION THREAD

**Discussion by AshZadePM** [noah-nuebling/mac-mouse-fix#432] [Questions] [answered]
I'd love to be able to go back and forward using horizontal mouse wheel buttons but it doesn't work in Safari - it starts the animation but then snaps back. In FF, if I hold it down, it eventually navigates back. If I hold Shift and scroll, I can navigate back and forward.

Here's Safari: 

https://user-images.githubusercontent.com/114925803/199282701-d99a5546-bbd5-46c1-80c1-4880de9dde96.mp4

Does anyone have any advice on how to troubleshoot this?

**Comment by noah-nuebling** [comment]
Hi thanks for the feedback!

I think this is a macOS bug. If you have a Trackpad or Magic Mouse see if it behaves the same way. 

Just restarting your computer might fix it

**Reply by AshZadePM** [reply]
Update: Beta 5 back works if I hold down the button. If I click it, it does the same thing as the recording I posted originally. Almost there!

**Reply by noah-nuebling** [reply]
Ohhh I think this might be a problem in the naming of the feature. The 'Click and Drag' features will only work if you hold down the button while you drag and the 'Click and Scroll' features only work if you hold down the button while you scroll.  

Is that where the confusion lies?

I chose that naming because "Click and Hold and then Scroll" would be too long but I understand how it can be confusing.

Thanks for the update!

**Reply by noah-nuebling** [reply]
Hey @AshZadePM, I think for some reason I just didn't get that you were trying to go back and forward by *tilting* your *scrollwheel*. Sorry about that. 

The problem is that most scrollwheels, when you tilt them, just send a sequence of scrollwheel events pretty far spaced out in time. Mac Mouse Fix sees this as you starting to scroll, but then stopping, and then starting again, and then stopping again, and so on. The events are too far spaced out to be picked up as a continuous series of scrolls that belong together. That's why going back to the previous page in Safari won't work properly by tiliting the scroll wheel.

The only proper way I can think of to fix this is to make Mac Mouse Fix reprogram your mouse to send distinct "scroll wheel was tilted left" "scroll wheel is no longer tilted left" events, instead of the mouse pretending like you're scrolling on a horizontal scrollwheel. Then Mac Mouse Fix would know exactly what you're doing with the scrollwheel and I could write some custom code into Mac Mouse Fix for properly handling scrollwheel tilting.

But this is a while off at this point, since it'll be a lot of work and reprogramming the mouse would work differently for each manufacturer and model.

Which mouse model are you using by the way?

Alternatively maybe you could get your hands on another driver software that can reprogram your mouse and increase the frequency of scroll events being sent when you tilt your wheel. Maybe if you play around with the smoothness and the scroll speed settings in MMF that could also help, but probably not.

Hope that makes some sense!

**Comment by AshZadePM** [comment]
Hello,

I restarted and it's still and issue. I don't have a Magic Mouse.

**Reply by noah-nuebling** [reply]
What happens if you use the 'Click and Drag' to 'Scroll & Navigate' feature in Mac Mouse Fix to go back and forward in Safari?

**Reply by AshZadePM** [reply]
That does work for me, but it's finicky. Like it doesn't work every third time. I may not be doing it properly.

**Comment by noah-nuebling** [comment, marked as answer]
I'm marking this discussion as answered since I'm moving it to https://github.com/noah-nuebling/mac-mouse-fix/issues/465.

Please continue any discussion there so I can keep an overview of all the issues and requests. Thank you!
